ps.go

```diff
@@ -73,28 +82,45 @@ var psCommand = cli.Command{
 				continue
 			}
 			fields := strings.Fields(line)
-			p, err := strconv.Atoi(fields[pidIndex])
-			if err != nil {
-				return fmt.Errorf("unexpected pid '%s': %s", fields[pidIndex], err)
-			}
-
-			for _, pid := range pids {
-				if pid == p {
-					fmt.Println(line)
-					break
-				}
+			if fields[pidIndex] != cmdPs {
+				fmt.Println(line)
 			}
 		}
 		return nil
-	},
+	}
+
+	ns := fmt.Sprintf("/proc/%d/ns/pid", pids[0])
+	fd, err := syscall.Open(ns, syscall.O_RDONLY, 0644)
+	if err != nil {
+		return fmt.Errorf("open /proc/%d/ns/pid %v", pids[0], err)
+	}
+	defer syscall.Close(fd)
+	err = system.Setns(uintptr(fd), 0)
```

Comment:
Ah, okay. This is how I would like this to be done, unfortunately there are several potential problems with trying to `Setns` in Go:

1. Since Go programs are multithreaded (with no way of disabling it), this syscall can fail arbitrarily (the interface says that you should only use `setns` for single-threaded programs).

2. In SELinux (and rootless container) contexts, you'll also need to join the user namespace as well.

I think this would work better if we made `libcontainer.Container.Processes()` do all of this internally, where it goes through the full `runc exec` process (but only joins the User and PID namespaces). Then we send the PIDs to the main runC process over an AF_UNIX socket.